src/frontend/src/redux/system/actions.ts

```diff
@@ -1,10 +1,21 @@
-import { LOG_IN, LOG_OUT, LogInData } from './types';
+import apiService from '../../service/api-service';
+import history from '../../service/history';
+import { LOG_IN, LOG_OUT } from './types';
 
```

Comment:
These changes look great!

Just so I understand, for a request that requires authentication we would include:
`authenticated: true` 
In the request, would we be pulling that value from the component and passing it in similar to:
`logIn(email, password, authenticated)`

Reply:
Whether a given request needs to be authenticated or not typically wouldn't change so it'd be declared statically in the action creator like e.g. this:

```
export function getRecords() {
  return (dispatch: Function) => {
    return apiService(dispatch, {
      url: '/api/records',
      method: 'get',
      authenticated: true
    }).then((response: any) => {
```

Reply:
@ForrestLonganecker @maxkwallace, would it make more sense to have the dispatch as the 2nd argument?  I would imagine that there are use cases where dispatch isn't required?  This would eliminate us needed to call apiService(null, request)

Reply:
@SteelFace87 if dispatch weren't required for every call to `apiService` it'd definitely be better for it to be an optional second argument. However that change can't be made without introducing other tradeoffs-- I hope my other comments above and my code comments in the PR make that clear. If you have a proposal for how to do this plus an argument for why it'd be better then I'm happy to hear it. 

Just to clarify, with the current implementation, one shouldn't do `apiService(null, request)` -- it won't work, since `dispatch` is used for every request regardless of whether it's authenticated or not.